config/global.ini.php

```diff
@@ -996,6 +996,11 @@
 ; For a list of tracking parameters you can use on the left side view https://developer.matomo.org/api-reference/tracking-api
 exclude_requests = ""
 
+; Custom image to return when tracker URL includes &image=1
+; Overrides the default 1x1 transparent gif
+; This should either be the full path to the image file or a base64 encoded image string wrapped in quotes
```

Comment:
do I see this right that it has to be a gif when it's a string? Might need to mention this?

Reply:
Yes that's true, that should at least be covered in the config notes. I don't want to determine the mime type using getimagesize() as it will require writing the string to a file first which will incur disk i/o making the more performant string option pointless. Other options would be to add an additional config option to set the mime type when a string is used (increases the possibility of mis-configuration) or add a method that attempts to determine the mime type by looking for gif/png/jpg signature bytes at the start of the string.

What do you think?

Reply:
getimagesizefromstring() looks like a good option for automatically determining the mime type from the string without incurring disk overhead.

Reply:
Change committed to automatically determine the mime type for image strings